### PR TITLE
Build AWS SDK as part of the prerequisites

### PR DESCRIPTION
**Pull request by bettinaheim**
Mostly incorporates https://github.com/NVIDIA/cuda-quantum/pull/2362, except for anyon related changes.

### CODE REVIEW

File: targettests/execution/bug_qubit.cpp [hunk 1]
```diff
@@ -10,11 +10,13 @@
...
 // clang-format off
 // RUN: nvq++ %cpp_std --target anyon                    --emulate %s -o %t && %t | FileCheck %s
-// RUN: nvq++ %cpp_std --target braket                   --emulate %s -o %t && %t | FileCheck %s
 // RUN: nvq++ %cpp_std --target ionq                     --emulate %s -o %t && %t | FileCheck %s
 // RUN: nvq++ %cpp_std --target iqm --iqm-machine Adonis --emulate %s -o %t && %t | FileCheck %s
 // RUN: nvq++ %cpp_std --target oqc                      --emulate %s -o %t && %t | FileCheck %s
 // RUN: nvq++ %cpp_std --target quantinuum               --emulate %s -o %t && %t | FileCheck %s
+// RUN: if $braket_avail; then \
+// RUN:   nvq++ %cpp_std --target braket                 --emulate %s -o %t && %t | FileCheck %s; \
+// RUN: fi
```
Comment (schweitzpgi) [inline targettests/execution/bug_qubit.cpp:19]: I don't like all this whitespace and extra, unnecessary formatting. It just makes it hard to cut and paste the command.

File: CMakeLists.txt [hunk 2]
```diff
@@ -162,13 +175,8 @@ if(NOT CURL_LIBRARY AND EXISTS "$ENV{CURL_INSTALL_PREFIX}/lib/libcurl.a")
...
 endif()
-if (NOT crypto_LIBRARY AND CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" AND EXISTS "$ENV{OPENSSL_INSTALL_PREFIX}/lib64/libcrypto.a")
-  SET(crypto_LIBRARY "$ENV{OPENSSL_INSTALL_PREFIX}/lib64/libcrypto.a" CACHE INTERNAL "")
-elseif(NOT crypto_LIBRARY AND NOT CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64" AND EXISTS "$ENV{OPENSSL_INSTALL_PREFIX}/lib/libcrypto.a")
-  SET(crypto_LIBRARY "$ENV{OPENSSL_INSTALL_PREFIX}/lib/libcrypto.a" CACHE INTERNAL "")
-endif()
-if (NOT crypto_INCLUDE_DIR AND EXISTS "$ENV{OPENSSL_INSTALL_PREFIX}/include")
-  SET(crypto_INCLUDE_DIR "$ENV{OPENSSL_INSTALL_PREFIX}/include" CACHE INTERNAL "")
+if(NOT AWSSDK_ROOT)
+  SET(AWSSDK_ROOT "$ENV{AWS_INSTALL_PREFIX}")
```
Comment (schweitzpgi) [inline CMakeLists.txt:179]: This approach seems disconnected from the real aim here. CMake should not be using magic environment settings that are set by the CI build scripts.

File: test/lit.site.cfg.py.in [hunk 3]
```diff
@@ -43,6 +43,13 @@ config.test_remote_sim = "@CUDAQ_TEST_REMOTE_SIM@"
 if cmake_boolvar_to_bool(config.test_remote_sim):
   config.available_features.add('remote-sim')
 
+config.cudaq_backends_braket = "@CUDAQ_ENABLE_BRAKET_BACKENDS@"
+if cmake_boolvar_to_bool(config.cudaq_backends_braket):
+  config.available_features.add('braket')
+  config.substitutions.append(('%braket_avail', 'true'))
+else:
+  config.substitutions.append(('%braket_avail', 'false'))
+
```
Comment (schweitzpgi) [inline test/lit.site.cfg.py.in:52]: There should never be executable tests in this directory, so this change isn't needed.

File: CMakeLists.txt (hunk 2 above)
Comment (bettinaheim) [reply CMakeLists.txt:179]: AWSSDK_ROOT needs to be defined for CMake to find the aws sdk. Setting environment variables for paths matches what we do for other components (e.g. cutensornet and custatevec libraries) and ensures that we can build in different environments. Keep in mind that cmake is not necessarily directly invoked, so passing this to cmake is not a great option. Additionally, CMake itself will actually pick this up from environment variables (if you define AWSSDK_ROOT as environment variable, CMake should pick that up as well).

File: targettests/execution/bug_qubit.cpp (hunk 1 above)
Comment (bettinaheim) [reply targettests/execution/bug_qubit.cpp:19]: $braket_avail is replaced by lit, so I don't think you would usually copy-paste the command including that. Hence, I personally find this easier to copy-paste and more readable. That said, I am happy to change it to be one line.

File: CMakeLists.txt (hunk 2 above)
Comment (schweitzpgi) [reply CMakeLists.txt:179]: I don't really agree.

cmake is invoked directly. I do it multiple times a day, in fact. I do not read through shell scripts to try and figure out special environment variables and would not expect anyone else to do so.

cmake should use find to determine if the aws sdk is installed and there should be some standardish cmake variable settings to override where find looks. e.g., `-DAWS_SDK_DIR=/path`.

Comment (bettinaheim) [reply CMakeLists.txt:179]: Discussed offline to change all *INSTALL_PREFIX variables to the *ROOT variable in a separate PR. 
Keeping the default configuration to build the AWS integration, since we don't have a full CI setup to ensure these pieces are included if the build too willingly just skips them. Any AWS components can be omitted by setting CUDAQ_ENABLE_BRAKET_BACKENDS to false. In that case, the everything works without the need to install the aws sdk beforehand.

File: CMakeLists.txt [hunk 4]
```diff
@@ -93,6 +93,11 @@ if (CUDAQ_ENABLE_REST AND NOT DEFINED CUDAQ_ENABLE_REMOTE_SIM)
   endif()
 endif()
 
+# Enable AWS backends by default.
+if (NOT DEFINED CUDAQ_ENABLE_BRAKET_BACKENDS)
+  set(CUDAQ_ENABLE_BRAKET_BACKENDS ON CACHE BOOL "Enable building AWS backends.")
```
Comment (schweitzpgi) [inline CMakeLists.txt:98]: Can we drop the plural `S` here?

File: targettests/execution/angled_gate.cpp [hunk 5]
```diff
@@ -6,7 +6,7 @@
  * the terms of the Apache License 2.0 which accompanies this distribution.    *
  ******************************************************************************/
 
-// RUN: nvq++ %cpp_std --target braket --emulate %s -o %t && %t | FileCheck %s
+// RUN: if $braket_avail; then nvq++ %cpp_std --target braket --emulate %s -o %t && %t | FileCheck %s; fi
```
Comment (annagrin) [inline targettests/execution/angled_gate.cpp:9]: Heads up: I added a couple of new braket tests in this PR: https://github.com/NVIDIA/cuda-quantum/pull/2431, so they would need to be updated as well if my PR merges first.

File: CMakeLists.txt (hunk 4 above)
Comment (bettinaheim) [reply CMakeLists.txt:98]: done

File: CMakeLists.txt [hunk 6]
```diff
@@ -93,6 +93,11 @@ if (CUDAQ_ENABLE_REST AND NOT DEFINED CUDAQ_ENABLE_REMOTE_SIM)
   endif()
 endif()
 
+# Enable AWS backends by default.
+if (NOT DEFINED CUDAQ_ENABLE_BRAKET_BACKEND)
+  set(CUDAQ_ENABLE_BRAKET_BACKEND ON CACHE BOOL "Enable building AWS backends.")
```
Comment (khalatepradnya) [inline CMakeLists.txt:98]: Suggest changing message to `Enable building Amazon Braket backends.`

**Review by khalatepradnya — APPROVED** [review]
👍
